python/cudf/cudf/tests/test_groupby.py

```diff
@@ -381,19 +381,76 @@ def emulate(df):
 
 
 @pytest.fixture(scope="module")
-def groupby_jit_data():
+def groupby_jit_data_small():
+    """
+    Small dataset for testing groupby apply with jit
+    returns a dataframe whose keys columns define
+    4 groups of size 1, 2, 3, 4 as well as an additional
+    key column that can be used to test subgroups
+
+    useful for very basic testing of result values
+
+    """
     np.random.seed(0)
     df = DataFrame()
-    nelem = 20
-    df["key1"] = np.random.randint(0, 3, nelem)
-    df["key2"] = np.random.randint(0, 2, nelem)
-    df["val1"] = np.random.random(nelem)
-    df["val2"] = np.random.random(nelem)
-    df["val3"] = np.random.randint(0, 10, nelem)
-    df["val4"] = np.random.randint(0, 10, nelem)
+    key1 = (
+        [1]  # group of size 1
+        + [2, 2]  # group of size 2
+        + [3, 3, 3]  # group of size 3
+        + [4, 4, 4, 4]  # group of size 4
+    )
```

Comment:
Writing it this way should make the group sizes more obvious.
```suggestion
    key1 = (
        [1]
        + [2] * 2
        + [3] * 3
        + [4] * 4
    )
```